crates/autopilot/src/decoded_settlement.rs

```diff
@@ -380,27 +357,24 @@ impl DecodedSettlement {
         order: &OrderExecution,
         trade: &DecodedTrade,
     ) -> Option<Fees> {
-        let solver_fee = match &order.executed_solver_fee {
-            Some(solver_fee) => *solver_fee,
-            None => {
-                // get uniform prices
-                let sell_index = self
-                    .tokens
-                    .iter()
-                    .position(|token| token == &order.sell_token)?;
-                let buy_index = self
-                    .tokens
-                    .iter()
-                    .position(|token| token == &order.buy_token)?;
-                let uniform_sell_price = self.clearing_prices.get(sell_index).cloned()?;
-                let uniform_buy_price = self.clearing_prices.get(buy_index).cloned()?;
-
+        let sell_index = trade.sell_token_index.as_u64() as usize;
```

Comment:
Together with the comment 👇 , I tried to find more readable code but failed. Note that most of the orders will go through the `ExecutedFee::Surplus` code path once swap with zero fees feature is enabled. Feel free to reorganize the code in another way in a separate PR, since I failed to make any improvements (or even in this one).